### PR TITLE
doc: various page improvements

### LINKED ISSUE #969: Remove depreacted/outdated platforms from documentation
Platforms like Blackberry10/BB10 or WP8 are still present in some parts of the current docs. Those should be removed:

https://cordova.apache.org/docs/en/latest/plugin_ref/spec.html#engines-and-engine
https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-device/index.html
https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-file/index.html

Child issue: https://github.com/apache/cordova-electron/issues/49

### LINKED ISSUE #1169: document missing android preferences
# Documentation Request

Add new `config.xml` preference `AndroidXAppCompatVersion` to document.

This preference is:

* Only for the `Android` patform
* Defaults set to `1.2.0` the current latest stable version of the `AppCompat` library.

Add a note that changing is at the users own risk.

### LINKED ISSUE #1255: Document new android preferences
# Feature Request

## Motivation Behind Feature

To document new preferences that was added in cordova-android.
Brought to our attention by: https://github.com/apache/cordova-android/pull/1410#issuecomment-1193351277

Most of the new preferences was introduced in https://github.com/apache/cordova-android/pull/1212

The current list of preferences for android can be found at (omitting the preferences that is already documented): https://github.com/apache/cordova-android/blob/60e3803c67734087b7a6ff8ee8bd3a659cb050f8/lib/prepare.js#L99-L111

```javascript
        { xmlKey: 'android-compileSdkVersion', gradleKey: 'COMPILE_SDK_VERSION', type: Number },
        { xmlKey: 'android-buildToolsVersion', gradleKey: 'BUILD_TOOLS_VERSION', type: String },
        { xmlKey: 'GradleVersion', gradleKey: 'GRADLE_VERSION', type: String },
        { xmlKey: 'AndroidGradlePluginVersion', gradleKey: 'AGP_VERSION', type: String },
        { xmlKey: 'AndroidXAppCompatVersion', gradleKey: 'ANDROIDX_APP_COMPAT_VERSION', type: String },
        { xmlKey: 'AndroidXWebKitVersion', gradleKey: 'ANDROIDX_WEBKIT_VERSION', type: String },
        { xmlKey: 'GradlePluginGoogleServicesVersion', gradleKey: 'GRADLE_PLUGIN_GOOGLE_SERVICES_VERSION', type: String },
        { xmlKey: 'GradlePluginGoogleServicesEnabled', gradleKey: 'IS_GRADLE_PLUGIN_GOOGLE_SERVICES_ENABLED', type: Boolean },
```

I'll list some suggested brief descriptions, which is not set in stone, and definitely not meant to be copied and pasted. They are just general highlights of what the preference is used for. Reviews can be done in any given PRs.

### android-compileSdkVersion

Defaults to `android-targetSdkVersion`. Sets the compile sdk version setting. Should find android doc link.

### android-buildToolsVersion

Defaults are cordova-android version dependent. Expects a full version string `eg. "32.0.0"`. Changing this may also requires changing the `PATH` environment variable to find the proper build tools.

This preference is primarily for cordova development, for testing upcoming versions of the Android SDK. Changing this has a high risk of breaking builds as newer build tools frequently introduce breaking changes.

### GradleVersion

Sets the gradle wrapper version to use. 

This preference is primarily for cordova development, for testing upcoming versions of the Android SDK. Changing this has a high risk of breaking builds as newer build tools frequently introduce breaking changes.

### AndroidGradlePluginVersion

Sets the Android Gradle Plugin version to use. 

This preference is primarily for cordova development, for testing upcoming versions of the Android SDK. Changing this has a high risk of breaking builds as newer build tools frequently introduce breaking changes.

### AndroidXAppCompatVersion

Someone more knowledgeable will have to fill in the description for me. (Maybe brief description of what the AndroidX App Compat is used for?)

### AndroidXWebKitVersion

Someone more knowledgeable will have to fill in the description for me. (Maybe brief description of what the AndroidX Webkit dependency is for?)

### GradlePluginGoogleServicesVersion

Overrides the google services library version.

### GradlePluginGoogleServicesEnabled

Enables Google Services. (I think google API keys are required?)

### PR DESCRIPTION
**Pull request by erisu**
### Motivation and Context

Various improvements on the config.xml page.

* Closes #1255
* Closes #1169
* Closes #969
  * In terms of `cordova-plugin-file`, this will be handled outside of the `cordova-docs` repo.

### Description

- Clarity
- Formatting
- Layout
  - Improved `plugin.xml` display for platform and variable tagging
- Wording
- Removed Deprecated Platform Information

### Testing

n/a

### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Review by dpogue — COMMENTED** [review]
The `Attributes(type) <br/> <span class="sub-header">Only for platform:</span> | Description` table headings should maybe just be `Attribute | Description` now?

File: www/docs/en/dev/guide/platforms/android/index.md [hunk 1]
```diff
@@ -31,19 +31,119 @@ Regardless of whether you intend to utilize Android-specific command-line tools
...
-6.X.X                   | 16 - 26                      | 4.1 - 8.0.0
-5.X.X                   | 14 - 23                      | 4.0 - 6.0.1
-4.1.X                   | 14 - 22                      | 4.0 - 5.1
-4.0.X                   | 10 - 22                      | 2.3.3 - 5.1
-3.7.X                   | 10 - 21                      | 2.3.3 - 5.0.2
+<table>
+    <thead>
+        <tr>
+            <th>cordova-android Version</th>
+            <th>Android API-Levels (Android Version)</th>
```
Comment (NiklasMerz) [inline www/docs/en/dev/guide/platforms/android/index.md:38]: Why replace the md table with a HTML table?

File: www/docs/en/dev/guide/platforms/ios/plugin.md [hunk 2]
```diff
@@ -260,8 +260,8 @@ For JavaScript, you can attach Safari to the app running within the iOS Simulato
 - Don't forget to add your plugin's mapping to `config.xml`. If you
   forget, an error is logged in the Xcode console.
 
-- Don't forget to add any hosts you connect to in the whitelist, as
-  described in Domain [Whitelist Guide](../../appdev/whitelist/index.html). If you forget, an error is
+- Don't forget to add any hosts you connect to in the allo list, as
```
Comment (NiklasMerz) [inline www/docs/en/dev/guide/platforms/ios/plugin.md:263]:
```suggestion
- Don't forget to add any hosts you connect to in the allow list, as
```

**Review by NiklasMerz — APPROVED** [review]
Looks good generally. Found some typos and did not build locally to test.

**Review by dpogue — APPROVED** [review]

File: www/docs/en/dev/plugin_ref/spec.md [hunk 3]
```diff
@@ -55,12 +55,12 @@ The child elements of the `<engines>` element specify versions of Apache Cordova
...
-scriptSrc(string) | **For custom frameworks only** <br/> *Required* <br/>  The script file that tells plugman the version of the custom framework. Ideally, this file should be within the top level directory of your plugin directory.
-platform(string) | **For custom frameworks only** <br/> *Required* <br/> The platforms your framework supports. You may use the wildcard `*` to say supported for all platforms, specify multiple with a pipe character like `android|ios` or just a single platform like `android`.
+name<br />{% cdv_vartype string %}| *Required* <br/>Name of the engine. Here are the default engines that are supported: <ul><li>`cordova`</li><li> `cordova-plugman`</li><li>`cordova-android`</li> <li>`cordova-browser`</li><li>`cordova-ios`</li><li>`windows-os`</li><li>`android-sdk` (returns the highest Android api level installed) </li><li> `apple-xcode` (returns the xcode version) </li><li>`apple-ios` (returns the highest iOS version installed)</li><li>`apple-osx` (returns the macOS version)</li></ul> You can also specify a custom framework apart from the default ones.
```
Comment (erisu) [inline www/docs/en/dev/plugin_ref/spec.md:60]:
```suggestion
name<br />{% cdv_vartype string %}| *Required* <br/>Name of the engine. Here are the default engines that are supported: `cordova`, `cordova-plugman`, `cordova-android`, `cordova-browser`, `cordova-ios`, `windows-os`, `android-sdk` (returns the highest Android api level installed) , `apple-xcode` (returns the xcode version), `apple-ios` (returns the highest iOS version installed), `apple-osx` (returns the macOS version). You can also specify a custom framework apart from the default ones.
```

**Pull request merged**